### PR TITLE
split playbook into roles, add local and remote zip-theme-file support

### PR DESCRIPTION
**Pull request by dgamanenko**
@imatviian please review

### CODE REVIEW

**Comment by imatviian** [comment]
@sidorovdmitry Please, review.

File: main.yml [hunk 1]
```diff
@@ -1,19 +1,65 @@
...
-        common_web_group: "www-data",
-        git_key: "{{ lookup('file', '~/.ssh/id_rsa_git') }}"
-
-      }
+  vars:
+    theme_dir_main: "/edx/app/edxapp/themes"
+    server_vars_path: '/edx/app/edx_ansible/server-vars.yml'
+    raccoongang_theme_dir_main: 'raccoongang_theme'
+    edxapp_data_dir: "/edx/app/edxapp"
+    git_ssh_key: "~/.ssh/id_rsa_git"
```
Comment (sydorovdmytro) [inline main.yml]: Move it to group_vars

File: roles/branch/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+---
+- name: Remove theme directory
+  file: 
+    path: "{{ theme_dir_main }}/{{ theme_branch }}" 
+    state: absent
+  become_method: sudo
+  become: yes
+  become_user: edxapp
```
Comment (sidorovdmitry) [inline roles/branch/tasks/main.yml]: Why remove? What we do if script fails? Maybe rename is better?

File: roles/branch/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+- include: private-repo-ssh.yml
+  when: ("{{ script_mode }}" == 'br')
+
+- name: Clone git theme repo
+  git: 
+    repo: "{{ theme_repo }}"
+    version: "{{ theme_branch }}"
+    dest: "{{ theme_dir_main }}/{{raccoongang_theme_dir_main}}/{{ theme_branch }}"
+    ssh_opts: "-o StrictHostKeyChecking=no"
+    accept_hostkey: yes
```
Comment (sydorovdmytro) [inline roles/branch/tasks/main.yml:19]: If you have `ssh_opts: "-o StrictHostKeyChecking=no"` option, `accept_hostkey: yes` will be ignored

File: roles/theme-zip/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    paths: "{{ theme_file_path }}"
+    patterns: "*.zip"
+  register: theme_zip
+  when: ("{{ script_mode }}" == 'l')
+
+- name: Get theme_folder_main name
+  #shell: 'find "{{ theme_file_path }}" -type f -name \*.zip | sed "s:.*/::" | sed "s/.zip//g"'
+  set_fact: 
+    theme_folder_main: "{{ raccoongang_theme_dir_main }}"
+  #when: ("{{ script_mode }}" == 'l')
```
Comment (sydorovdmytro) [inline roles/theme-zip/tasks/main.yml]: Remove unused comments

File: roles/theme-zip/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+- name: Remove theme directory
+  file:
+    path: "{{ theme_dir_main }}/{{theme_folder_main}}"
+    state: absent
+  when: (theme_folder_main is defined)
+  become_method: sudo
+  become: yes
+  become_user: edxapp
```
Comment (sidorovdmitry) [inline roles/theme-zip/tasks/main.yml]: Again. Why remove?

File: update_theme.sh [hunk 6]
```diff
@@ -1,18 +1,64 @@
...
+	elif test "$#" -eq 3;
+	then
+		if [ "$1" == "$mode_branch_custom_repo" ];
+		then
+			~/.edxthemevenv/bin/ansible-playbook -c local -i ",localhost" main.yml -e "mode=br" -e "theme_branch=$2 theme_repo=$3"
+		fi
+	elif test "$#" -eq 4;
+	then
+		if [ "$1" == "$mode_branch_custom_repo_remote_edx_host" ];
+		then
```
Comment (sydorovdmytro) [inline update_theme.sh:55]: Why not if $1 in the array of possible options, do 

`~/.edxthemevenv/bin/ansible-playbook -c local -i ",localhost" main.yml -e "mode=$1 local_zip=$2"` ?

File: roles/branch/tasks/main.yml (hunk 3 above)
Comment (dgamanenko) [reply roles/branch/tasks/main.yml:19]: fixed

File: update_theme.sh (hunk 6 above)
Comment (dgamanenko) [reply update_theme.sh:55]: fixed

**Pull request merged**